engine/src/main/java/org/terasology/i18n/assets/Translation.java

```diff
@@ -0,0 +1,116 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.i18n.assets;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.function.Consumer;
+
+import org.terasology.assets.Asset;
+import org.terasology.assets.AssetType;
+import org.terasology.assets.ResourceUrn;
+import org.terasology.engine.Uri;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * Defines a translation asset.
+ */
+public class Translation extends Asset<TranslationData> {
+
+    private final List<Consumer<Translation>> changeListeners = new CopyOnWriteArrayList<>();
+
+    private TranslationData data;
+    private Locale locale;
+    private Uri projectUri;
+
+    /**
+     * @param urn       The urn identifying the asset. Never <code>null</code>.
+     * @param assetType The asset type this asset belongs to. Never <code>null</code>.
+     * @param data      The actual translation data. Never <code>null</code>.
+     */
+    public Translation(ResourceUrn urn, AssetType<?, TranslationData> assetType, TranslationData data) {
+        super(urn, assetType);
+        reload(data);
+    }
+
+    /**
+     * @return the uri of the project this instance is part of
+     */
+    public Uri getProjectUri() {
+        return projectUri;
+    }
+    /**
+     * @return the locale of the translation data
+     */
+    public Locale getLocale() {
+        return locale;
+    }
+
+    /**
+     * Subscribe to reload/dispose events.
+     * @param changeListener the listener to add
+     */
+    public void subscribe(Consumer<Translation> changeListener) {
+        changeListeners.add(changeListener);
+    }
+
+    /**
+     * Unsubscribe from reload/dispose events.
+     * @param changeListener the listener to remove. Non-existing entries will be ignored.
+     */
+    public void unsubscribe(Consumer<Translation> changeListener) {
+        changeListeners.remove(changeListener);
+    }
+
+    /**
+     * Retrieves the matching entry for the given key.
+     * @param id the id of the entry
+     * @return the translated string
+     */
+    public String lookup(String id) {
+        return data.getTranslations().get(id);
+    }
+
+    @Override
+    protected void doDispose() {
+        this.data = null;
+        // locale and project uri are still available!
+        for (Consumer<Translation> listener : changeListeners) {
+            listener.accept(this);
+        }
+    }
+
+    @Override
+    public String toString() {
+        return getUrn().toString();
+    }
+
+    @Override
+    protected void doReload(TranslationData newData) {
+        Preconditions.checkArgument(newData != null);
+        this.data = newData;
```

Comment:
Suggest copy of data, or not retaining the data and inlining the details in the asset. Leaves you at risk of the data being changed outside of the proper reload pattern.